packages/nimble-components/src/theme-provider/design-tokens.ts

```diff
@@ -646,6 +699,55 @@ function hexToRgbPartial(hexValue: string): string {
     return `${red}, ${green}, ${blue}`;
 }
 
+function createDropShadowTokens(
+    dropShadowTokenName: string,
+    colorFunction: (element: HTMLElement) => string,
+    offsetX: string,
+    offsetY: string,
+    blur: string
+): readonly [
+        CSSDesignToken<string>,
+        CSSDesignToken<string>,
+        CSSDesignToken<string>,
+        CSSDesignToken<string>
+    ] {
+    if (
+        dropShadowTokenName === ''
+        || offsetX === ''
+        || offsetY === ''
+        || blur === ''
+    ) {
+        throw new Error(
+            'createDropShadowTokens parameter unexpectedly set to empty string'
+        );
+    }
+
+    const dropShadowToken = DesignToken.create<string>(
+        styleNameFromTokenName(dropShadowTokenName)
+    ).withDefault(
+        (element: HTMLElement) => `${offsetX} ${offsetY} ${blur} ${colorFunction(element)}`
+    );
+
+    const dropShadowColorToken = DesignToken.create<string>(
+        styleNameFromTokenName(`${dropShadowTokenName}-color`)
+    ).withDefault((element: HTMLElement) => colorFunction(element));
+
+    const dropShadowOffsetYToken = DesignToken.create<string>(
+        styleNameFromTokenName(`${dropShadowTokenName}-offset-y`)
+    ).withDefault(`${offsetY}`);
+
+    const dropShadowBlurToken = DesignToken.create<string>(
+        styleNameFromTokenName(`${dropShadowTokenName}-blur`)
+    ).withDefault(`${blur}`);
+
+    return [
+        dropShadowToken,
+        dropShadowColorToken,
+        dropShadowOffsetYToken,
```

Comment:
I dunno if it intentional but I think we should breakout all the parts of the top shadow token, so we should go ahead and do offset x as well